src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -209,33 +223,39 @@ def handle_warning_click(self, button):
         self.data_submitted = "OK" in button.text()
 
     def reduce_dimensions(self, dim_number=None):
-        self.manual_reduction.clicked.disconnect()
-        self.auto_reduction.clicked.disconnect()
-        self.engine.reduce_dimensions(dim_number)
+        self.engine.state.imported_data.drop(self.reduced_columns, axis=1, inplace=True, errors='ignore')
+        self.engine.state.raw_data.drop(self.reduced_columns, axis=1, inplace=True)
+        self.reduced_columns = self.engine.reduce_dimensions(dim_number)
```

Comment:
maybe here we could automatically select those reduced_columns (and unselect the rest)?